Zeek 6.1 doc updates (#217)
zeek/zeek-docs: script-reference/statements.rst

  * - :zeek:keyword:`add`, :zeek:keyword:`delete`
    - Add or delete elements

  * - :zeek:keyword:`assert`

>     - Runtime sanity checking

Reviewer: Some consider "sanity check" abelist, maybe go with something like _assertion_ instead. Here and below.

```suggestion
    - Runtime assertion
```

Author: I was not aware, thanks.

    add myset["test"];

.. zeek:keyword:: assert

assert
~~~~~~

.. versionadded:: 6.1

The ``assert`` statement can be used for runtime assertion checks or as a building
block for a testing framework. It takes an expression ``expr`` of type
:zeek:see:`bool` and an optional message of type :zeek:see:`string`.

> If ``expr`` at runtime evaluates to ``F``, by default the string representation
> of the expression and the given message, if any, are logged
> via :zeek:see:`Reporter::error`.

Reviewer 2: ```suggestion
If ``expr`` at runtime evaluates to ``F``, the string representation
of the expression and the given message, if any, are logged
via :zeek:see:`Reporter::error` by default.
```

Script execution for a given even handler stops with a failing ``assert`` statement
comparable to a scripting runtime error after generating the log.

Example:

.. literalinclude:: assert_1.zeek
   :language: zeek
   :linenos:
   :tab-width: 4

This script prints the following messages to stderr, as well logging them to
``reporter.log``.

.. code-block:: console

   $ zeek assert_1.zeek
   error in ./assert_1.zeek, line 6: assertion failure: 40 < x
   error in ./assert_1.zeek, line 12: assertion failure: 40 < x (37 is not greater than 40)

.. note::

   Zeek's exit code in this example will be ``0``, indicating success.
   Script errors other than those in a ``zeek_init()`` handler are not
   reflected in Zeek's exit code.

The logging behavior of failing assert statements can be customized using the
:zeek:see:`assertion_failure` or zeek:see:`assertion_result` hook.

> Using :zeek:see:`break` in either hook allows to suppress the default log
> generation. The :zeek:see:`assertion_result` hook is targeted for testing

Reviewer 2: ```suggestion
Using :zeek:see:`break` in either hook allows for suppression of the default log
generation. The :zeek:see:`assertion_result` hook is targeted for testing
```
